### DISCUSSION TITLE
Larger space

### DISCUSSION THREAD

**Discussion by Tamnac** [Lisnoti/Lisnoti#2] [Ideas]
I've been trialing Lisnoti for a while. It's a unique experience using a proportional font. Some nice things for sure. Also some annoyances.

I'd say the biggest annoyance right now is how small the spaces are. This means indenting is never enough. Now if you indent with tabs (which I do), you can always change the tab display size, however most people don't.

In some cases, it can also be a bit hard to tell whether there is a space or not, e.g

![Screenshot 2024-07-23 112321](https://github.com/user-attachments/assets/3c83a86b-532a-4930-bb7e-4f9ccd9c36ba)

I think it might be helpful to make the space like 40% or so larger.

**Comment by TimGord** [comment]
Thank you again for the feedback – it’s really useful.

The width of spaces is inherited from Noto Sans and is designed for general typesetting.

But I agree that it *is* an issue for reading code. In fact, I got bitten by this myself last week.

Leave it with me and I’ll try creating a version with wider spaces (possibly specifically for coding).

I’d also be interested in understanding the other annoyances you mentioned.

Cheers

**Reply by TimGord** [reply]
Based on your feedback, I’ve added an experimental version of the font with 40% wider spaces called [‘Lisnoti Code WS’](https://github.com/Lisnoti/Lisnoti/).

Space-based indentation using this version *is* clearer although, to my eyes at least, spacing now looks off for normal prose text. That said, I’m going to give it a go myself for coding and see how I get on.

For the record, I also tried a different approach, using kerning to make consecutive spaces much wider. While great in theory − because it preserves normal prose layout while also improving indentation − in practice it depends on apps obeying it and, based on a quick trial with VS Code and Visual Studio, this is not the case. (Visual Studio’s behaviour seemed to vary line-by-line.) So I’ve abandoned this approach for the time being.

**Reply by Tamnac** [reply]
40% was just a random number haha, I have no idea if it's ideal. 

So I tried it, some observations:

1. It does seem better
2. Doesn't look too bad for regular prose, I stopped noticing extra space fairly fast. It reminds of justified text, since that is usually more widely spaced.
3. vscode is bugging for some reason, I can't get it to work with the new one. Notepad++ worked fine. MS Word also worked fine. 
4. Yeah I don't think kerning is the solution, since it doesn't solve the ambiguous space problem I showed in the OP.

It's unfortunate, vscode is my main editor, so it'll be harder to test unless I can get it working

**Reply by io-canopy** [reply]
Why was Lisnoti Code WS removed? I am experiencing the same issues as OP.

**Reply by TimGord** [reply]
My apologies -- I didn't think it worked very well typographically and I wasn't aware anyone was using it.

I'll restore it later today.

Cheers

**Reply by TimGord** [reply]
I've restored Lisnoti Code WS.

If you have any feedback, please let me know.
